viite-UI/src/view/ProjectForm.js

```diff
@@ -340,31 +352,63 @@
         createOrSaveProject();
       };
 
-      var nextStage = function () {
+      var nextStage = function (reOpenCurrent = false, generalNext = false, saveAndNext = false) {
         applicationModel.addSpinner();
         currentProject.isDirty = false;
         jQuery('.modal-overlay').remove();
-        eventbus.trigger('roadAddressProject:openProject', currentProject);
+        if (generalNext || saveAndNext) {
+          eventbus.trigger('roadAddressProject:openProject', currentProject);
+        }
         rootElement.html(selectedProjectLinkTemplateDisabledButtons(currentProject));
         var projectErrors = projectCollection.getProjectErrors();
         // errorCode 8 means there are projectLinks in the project with status "NotHandled"
         var highPriorityProjectErrors = projectErrors.filter((error) => error.errorCode === 8);
+        if (generalNext || saveAndNext) { // if the project was opened or project was edited and opened again
+          buttonsWhenOpenProject(projectErrors, highPriorityProjectErrors);
+        } else if (reOpenCurrent) { // if project link form was closed without saving
+          buttonsWhenReOpenCurrent(projectErrors, highPriorityProjectErrors);
+        }
+        _.defer(function () {
+          applicationModel.selectLayer('roadAddressProject');
+          toggleAdditionalControls();
+        });
+      };
+
+      /**
+       * Set recalculate and changes buttons attributes (disabled, title) when project is opened
```

Comment:
Kielenhuolto:

"Set recalculate and changes buttons' attributes"  (<- tossa siishipsu lisää. )
Meni pitkä tovi lausetta parsiessa, kun en tajunnut recalculaten liittyvän buttoniin ja attribuutteihin, kun hipsu puuttui... 😁 
tai sit
"attributes of the recalculate, and changes buttons"

"**the** project".